chain-impl-mockchain/src/certificate/private_vote_tally.rs

```diff
@@ -0,0 +1,130 @@
+use crate::certificate::{TallyProof, VoteTallyPayload};
+use crate::{
+    certificate::{CertificateSlice, VotePlanId},
+    transaction::{Payload, PayloadAuthData, PayloadData, PayloadSlice},
+    vote::{PayloadType, TryFromIntError},
+};
+use chain_core::{
+    mempack::{ReadBuf, ReadError, Readable},
+    property,
+};
+use typed_bytes::{ByteArray, ByteBuilder};
+
+#[derive(Debug, Eq, PartialEq, Hash, Clone)]
+pub struct PrivateVoteTally {
```

Comment:
I think this could've been just `VoteTally` where `VoteTallyPayload` is
```rust
enum VoteTallyPayload {
    Public,
    Private { shares: Vec<Vec<chain_vote::TallyDecryptShare>> },
}
```
Although maybe I lost some part of the discussion. But again, what's the point of `VoteTallyPayload` if it can only be `Private`?